src/com/projectkorra/projectkorra/firebending/FireBlast.java

```diff
@@ -69,13 +72,23 @@ public FireBlast(final Location location, final Vector direction, final Player p
 
 		this.setFields();
 		this.safeBlocks = safeBlocks;
-		this.damage = damage;
 
 		this.location = location.clone();
 		this.origin = location.clone();
 		this.direction = direction.clone().normalize();
-		this.range = this.getDayFactor(this.range);
-		this.damage = this.getDayFactor(damage);
+
+		int damageMod = 0;
+		int rangeMod = 0;
+
+		damageMod = (int) (this.getDayFactor(damage) - damage);
+		rangeMod = (int) (this.getDayFactor(this.range) - this.range);
+
+		damageMod = (int) (bPlayer.canUseSubElement(SubElement.BLUE_FIRE) ? (BlueFireAbility.getDamageFactor() * damage - damage) + damageMod : damageMod);
+		rangeMod = (int) (bPlayer.canUseSubElement(SubElement.BLUE_FIRE) ? (BlueFireAbility.getRangeFactor() * range - range) + rangeMod : rangeMod);
+
+		this.range += rangeMod;
+		this.damage += damageMod;

```

Comment:
Why not simply times damage and range by either the blue fire multiplier, or 1 if they don't have blue fire? This seems so much more pointlessly complex.

Reply:
Because I'm trying to get rid of the multiplicative modifier and instead have an additive modifier. 

Previously the multipliers would either replace each other or compound on each other, so lets say I apply the day modifier to the range of fireblast, the modifier being 1.25 and the range being 10 blocks. The new range is now 12.5 blocks. Now if I were to solely apply the blue fire modifier which for purposes lets say is 1.5, it would be 12.5 * 1.5 = 18.75 blocks. With my changes, instead of this compounding each modifier what it does is it gets a total modifier and adds it to the base like so

Lets take the same values as above but apply it with the new formula

10 * 1.25  - 10 = 2.5
10 * 1.5 - 10 = 5

10 + 2.5 + 5 = 17.5

The difference is more noticeable as these modifiers get larger

10 * 1.5 = 15
15 * 1.5 = 22.5

10 * 1.5 - 10 = 5
10 * 1.5 - 10 = 5
10 + 10 = 20

And while for range values this may not be that bad, for the purposes of doing damage its much better this way since you dont need to worry about the modifiers compounding and you can solely focus on each modifiers boost individually.

Reply:
Alright. Totally agree with you. Always thought it was a problem too but wasn't sure the best way to solve it.

May be worth putting comments in to explain it though. Even just briefly.

Reply:
Good note!

Reply:
Another thought I had was you could make a generic multiplier method to make it look more clean. E.g. 

`double blueFireModifier = bPlayer.canUseSubElement(SubElement.BLUE_FIRE) ? BlueFireAbility.getDamageFactor() : 1;`
`this.damage = GeneralMethods.addMultipliers(damage, blueFireModifier, this.getDayFactor());`

If you made the method have a `double... modifiers` parameter, you could put as many as you need into it. Would certainly tidy up a lot of fire abilities doing the same thing.